### PR TITLE
DrawTool issues

### PR DESCRIPTION
**Pull request by chrisvanmook**

### CODE REVIEW

File: src/app/pages/MapPage/draw/DrawTool.tsx [hunk 1]
```diff
@@ -214,66 +212,68 @@ const DrawTool: FunctionComponent<DrawToolProps> = ({ setCurrentOverlay }) => {
...
+      if (layersToRemove.includes(typedLayer.id) && typedLayer instanceof Polygon) {
+        removeDataSelection()
+      }
+      drawnItemsGroup.removeLayer(typedLayer)
+    })
+
+    // Add layers if they don't exist on the map
+    const drawnItems: Array<PolylineType | PolygonType> = []
+    if (polygon) {
+      drawnItems.push(createPolyLayer(polygon))
```
[the remark this answers is not in the document]
Comment (alimpens) [reply src/app/pages/MapPage/draw/DrawTool.tsx:261]: Good enough reason for me

File: src/app/pages/MapPage/draw/DrawTool.test.tsx [hunk 2]
```diff
@@ -0,0 +1,11 @@
+describe('DrawTool', () => {
```
Comment (jonkoops) [inline src/app/pages/MapPage/draw/DrawTool.test.tsx:1]: Is there a ticket to implement these tests?

File: src/app/utils/useParam.ts [hunk 3]
```diff
@@ -28,23 +28,22 @@ const useParam = <T>(urlParam: UrlParam<T>): [T, SetValueFn<T>] => {
...
   }, [stateRef, state])
 
+  useEffect(() => {
+    locationRef.current = location
+  }, [locationRef, location])
+
   const setValue = useCallback<SetValueFn<T>>((valueOrFn, method = 'push') => {
     const value = valueOrFn instanceof Function ? valueOrFn(stateRef.current) : valueOrFn
-    const newValue = value ? encodeParam(urlParam, value) : null
+    const newValue = value !== null ? encodeParam(urlParam, value) : null
```
Comment (jonkoops) [inline src/app/utils/useParam.ts:43]: Should we not be handling this in `encodeParam` instead?

**Review by jonkoops — CHANGES_REQUESTED** [review]

File: package.json [hunk 4]
```diff
@@ -50,7 +50,7 @@
     "json-api-normalize": "1.1.5",
     "keycloak-js": "^12.0.3",
     "leaflet": "1.7.1",
-    "leaflet-draw": "1.0.0",
+    "leaflet-draw": "1.0.4",
```
[the remark this answers is not in the document]
Comment (chrisvanmook) [reply package.json:53]: It does, will downgrade 👍

**Review by jonkoops — APPROVED** [review]

**Pull request merged**